[CI] StatelessRealTimeGetIT testDataVisibility failing

**Build Scans:**
- [elasticsearch-intake #40352 / part2](https://gradle-enterprise.elastic.co/s/xt4os3exjbazy)
- [elasticsearch-pull-request #149701 / part-2](https://gradle-enterprise.elastic.co/s/3o3wy3kxrkygy)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:stateless:internalClusterTest" --tests "org.elasticsearch.xpack.stateless.StatelessRealTimeGetIT.testDataVisibility" -Dtests.seed=5E01F85BBD00C091 -Dtests.locale=lg -Dtests.timezone=Pacific/Kanton -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.stateless.StatelessRealTimeGetIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testDataVisibility'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: All incoming requests on node [node_t3] should have finished. Expected 0 bytes for requests in-flight but got 87 bytes; pending tasks [[{
  "node" : "node_t3",
  "id" : 12546,
  "type" : "transport",
  "action" : "indices:data/read/get[s]",
  "description" : "",
  "start_time" : "2026-05-27T19:04:48.229Z",
  "start_time_in_millis" : 1779908688229,
  "running_time" : "3.8m",
  "running_time_in_nanos" : 228924747064,
  "cancellable" : false,
  "headers" : { }
}]]
```

**Issue Reasons:**
- [main] 1 consecutive failures in step part2
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 2 failures in test testDataVisibility (0.5% fail rate in 414 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

There's an [exception in the test logs](https://gradle-enterprise.elastic.co/s/xt4os3exjbazy/tests/task/:x-pack:plugin:stateless:internalClusterTest/details/org.elasticsearch.xpack.stateless.StatelessRealTimeGetIT/testDataVisibility?expanded-stacktrace=WyIwIl0&top-execution=1), one of the [async threads](https://github.com/elastic/elasticsearch/blob/ea95ea832688adb4a8bbd506f00542810e036ac2/x-pack/plugin/stateless/src/internalClusterTest/java/org/elasticsearch/xpack/stateless/StatelessRealTimeGetIT.java#L421-L429) perhaps

```
Maa 28, 2026 8:04:48 AM com.carrotsearch.randomizedtesting.RandomizedRunner$QueueUncaughtExceptionsHandler uncaughtException	
WARNING: Uncaught exception in thread: Thread[#994,elasticsearch[node_t3][stateless_shard_read][T#3],5,TGRP-StatelessRealTimeGetIT]	
java.lang.AssertionError: elasticsearch[node_t3][stateless_shard_read][T#3] not in [stateless_fill_vbcc_cache, stateless_fill_vbcc_cache] nor a test thread	
	at __randomizedtesting.SeedInfo.seed([5E01F85BBD00C091]:0)	
	at org.elasticsearch.threadpool.ThreadPool.assertCurrentThreadPool(ThreadPool.java:1115)	
	at org.elasticsearch.xpack.stateless.cache.reader.SequentialRangeMissingHandler.lambda$fillCacheRange$0(SequentialRangeMissingHandler.java:200)	
	at org.elasticsearch.action.ActionListenerImplementations$MappedActionListener.lambda$map$0(ActionListenerImplementations.java:124)	
	at org.elasticsearch.action.ActionListenerImplementations$MappedActionListener.lambda$map$0(ActionListenerImplementations.java:124)	
	at org.elasticsearch.action.ActionListenerImplementations$MappedActionListener.onResponse(ActionListenerImplementations.java:105)	
	at org.elasticsearch.action.ActionRunnable$2.accept(ActionRunnable.java:58)	
	at org.elasticsearch.action.ActionRunnable$2.accept(ActionRunnable.java:55)	
	at org.elasticsearch.action.ActionRunnable$4.doRun(ActionRunnable.java:101)	
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)	
	at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:35)	
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:1114)	
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)	
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)	
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)	
	at java.base/java.lang.Thread.run(Thread.java:1516)
```

`stateless_shard_read` is the `SHARD_READ_THREAD_POOL` thread pool
```
    public static final String STATELESS_SHARD_READ_THREAD_NAME = "stateless_shard_read";
```
```
    public static final String SHARD_READ_THREAD_POOL = BlobStoreRepository.STATELESS_SHARD_READ_THREAD_NAME;
```

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -274,9 +274,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecWhereLikeIT
   method: test {csv-spec:where-like.likePrefixSuffix}
   issue: https://github.com/elastic/elasticsearch/issues/149964
-- class: org.elasticsearch.xpack.stateless.StatelessRealTimeGetIT
-  method: testDataVisibility
-  issue: https://github.com/elastic/elasticsearch/issues/150028
 - class: org.elasticsearch.xpack.esql.CsvIT
   method: test {csv-spec:k8s-timeseries-increase.increase_of_double_no_grouping_promql}
   issue: https://github.com/elastic/elasticsearch/issues/150045

